fix : added tooltip for column fields

musharaf-nocodb opened the pull request:
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

https://www.loom.com/share/675e1c52230d4a0fbd0b03bc84770d4c?sid=3f33b8ae-6863-4219-90cb-d43ab09d6bff

![image](https://github.com/nocodb/nocodb/assets/152274770/4c01a3a7-fcd9-4fcb-b30a-22458e7f640c)


**Added ToolTip in below sections**

- [ ] Column fields in View Row Section
- [ ] Filters Column Fields
- [ ] Sidebar Tables and Views names
- [ ] Topbar Views names
- [ ] Child table names
- [ ] Fixed width in ERD column Fields 
- [ ] Sort Fields
- [ ] UI-ACL header and cell Values

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

dstala commented:
- [ ] Needs fix in below highlighted areas
![Screenshot 2023-11-29 at 9 11 55 PM](https://github.com/nocodb/nocodb/assets/86527202/94860ee7-72ba-4400-acab-a8d36704e802)

- [ ] In toolbar filter, group-by, sort menu :: let us limit width, show `...` by default & show full name as tooltip on hover
![Screenshot 2023-11-29 at 9 15 14 PM](https://github.com/nocodb/nocodb/assets/86527202/229d944f-7590-4bd6-8265-cea66776f00f)

- [ ] Needs fix in details tab
  - display field should be truncated
  - rest of the fields are truncated, but doesn't display full name on hover
 
![Screenshot 2023-11-29 at 9 16 06 PM](https://github.com/nocodb/nocodb/assets/86527202/cffd342c-4c32-4e97-9f3d-e0a820e59b5a)

mustafapc19 commented on packages/nc-gui/components/smartsheet/details/Fields.vue:
```diff
@@ -814,14 +814,16 @@ const onFieldOptionUpdate = () => {
-                    >
-                      {{ fieldState(field)?.title || field.title }}
-                    </span>
+                    <a-tooltip :title="fieldState(field)?.title || field.title">
```
Use `NcTooltip` component, `packages/nc-gui/components/nc/Tooltip.vue`

musharaf-nocodb replied:
**Fixed in below commit**

[fix : Used NcTooltip](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)
[9207eda](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)

mustafapc19 commented on packages/nc-gui/components/smartsheet/header/Cell.vue:
```diff
@@ -90,18 +90,20 @@ const onClick = (e: Event) => {
-    >
-      {{ column.title }}
-    </div>
+    <a-tooltip placement="bottom" :title="column.title">
```
Use NcTooltip component

musharaf-nocodb replied:
**Fixed in below commit**

[fix : Used NcTooltip](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)
[9207eda](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)

mustafapc19 commented on packages/nc-gui/components/smartsheet/header/VirtualCell.vue:
```diff
@@ -153,11 +153,11 @@ const openDropDown = (e: Event) => {
     <LazySmartsheetHeaderVirtualCellIcon v-if="column && !props.hideIcon" />
 
-    <a-tooltip placement="bottom">
+    <a-tooltip placement="bottom" :title="column.title">
```
Use NcTooltip component

musharaf-nocodb replied:
**Fixed in below commit**

[fix : Used NcTooltip](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)
[9207eda](https://github.com/nocodb/nocodb/pull/7114/commits/9207eda4c98368d9b747fb3ccff4a37b70b60fc7)

mustafapc19 commented on packages/nc-gui/components/smartsheet/header/VirtualCell.vue:
```diff
@@ -153,11 +153,11 @@ const openDropDown = (e: Event) => {
         {{ tooltipMsg }}
       </template>
-      <span class="name truncate pl-1" :class="{ truncate: !isForm }" :data-test-id="column.title">
+      <span class="name pl-1" :class="{ truncate: !isForm }" :data-test-id="column.title">
```
Is removing truncation for form intentional?

musharaf-nocodb replied:
**fixed in below commit**

[fix : fixed tooltip for virtual cell](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)
[e84c468](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)

mustafapc19 commented:
Expanded form used to have left and right padding which seems to be removed. Intentional?
New UI:
<img width="1515" alt="image" src="https://github.com/nocodb/nocodb/assets/35809690/0ec8b227-7507-4509-9192-87e837f44d8d">
Old UI:
<img width="1493" alt="image" src="https://github.com/nocodb/nocodb/assets/35809690/d29c97d6-543a-4f84-b3f3-0f400515022e">

wingkwong commented on packages/nc-gui/components/smartsheet/column/LinkedToAnotherRecordOptions.vue:
```diff
@@ -74,16 +74,18 @@ const isLinks = computed(() => vModel.value.uidt === UITypes.Links)
           show-search
           :filter-option="filterOption"
-          dropdown-class-name="nc-dropdown-ltar-child-table"
+          dropdown-class-name="nc-dropdown-ltar-child-table "
```
unexpected space. please run lint.

musharaf-nocodb replied:
**fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

wingkwong commented on packages/nc-gui/components/smartsheet/header/Cell.vue:
```diff
@@ -90,18 +90,27 @@ const onClick = (e: Event) => {
+
+      <div
+        :class="{
+          '!truncate': !isForm,
```
1. `!` not required.
2. since you change to `NcTooltip in L93, the truncate in L98 is no longer required.

wingkwong replied:
Got it. I think `!` is not required for both place tho. Please double check.

musharaf-nocodb replied:
1.) **fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

musharaf-nocodb replied:
! removed in all places

wingkwong commented on packages/nc-gui/components/smartsheet/details/Fields.vue:
```diff
@@ -1072,7 +1079,7 @@ const onFieldOptionUpdate = () => {
           </div>
           <Transition v-if="!changingField" name="slide-fade">
-            <div class="border-gray-200 border-l-1 rounded-r-xl h-[calc(100vh-(var(--topbar-height)*3.85))]">
+            <div class="border-gray-200 border-l-1 rounded-r-xl nc-scrollbar-md nc-fields-height !overflow-y-auto">
```
Please explain why these classes are required.

musharaf-nocodb replied:
**Removed Unwanted classes in below commit** 

[fix : fixed overflow in fieldContainer](https://github.com/nocodb/nocodb/pull/7114/commits/a31b544ba56eac7876bf1e428bac3e1e4000a298)
[a31b544](https://github.com/nocodb/nocodb/pull/7114/commits/a31b544ba56eac7876bf1e428bac3e1e4000a298)

wingkwong commented on packages/nc-gui/components/smartsheet/expanded-form/index.vue:
```diff
@@ -632,24 +630,21 @@ export default {
                   <LazySmartsheetHeaderVirtualCell
                     v-if="isVirtualCol(col)"
-                    class="nc-expanded-cell-header h-full"
+                    class="nc-expanded-cell-header h-full !max-w-80"
```
don't misuse `!`. You don't need that here.

musharaf-nocodb replied:
**fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

mustafapc19 commented on packages/nc-gui/components/erd/TableNode.vue:
```diff
@@ -54,7 +54,7 @@ watch(
 
     <div
-      class="relative h-full flex flex-col justify-center bg-white min-w-16 min-h-8 rounded-lg nc-erd-table-node"
+      class="relative h-full max-w-[20rem] flex flex-col justify-center bg-white min-w-16 min-h-8 rounded-lg nc-erd-table-node"
```
Better to use standard tailwind class for`max-w-[20rem]` which would be `w-80`

musharaf-nocodb replied:
**fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

wingkwong commented on packages/nc-gui/components/smartsheet/expanded-form/index.vue:
```diff
@@ -632,24 +630,21 @@ export default {
                   />
 
-                  <LazySmartsheetHeaderCell v-else class="nc-expanded-cell-header" :column="col" />
+                  <LazySmartsheetHeaderCell v-else class="nc-expanded-cell-header !max-w-80" :column="col" />
```
same as above

musharaf-nocodb replied:
**fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

mustafapc19 commented on packages/nc-gui/components/smartsheet/column/LinkedToAnotherRecordOptions.vue:
```diff
@@ -74,16 +74,18 @@ const isLinks = computed(() => vModel.value.uidt === UITypes.Links)
           show-search
           :filter-option="filterOption"
-          dropdown-class-name="nc-dropdown-ltar-child-table"
+          dropdown-class-name="nc-dropdown-ltar-child-table "
```
Extra space here. Intentional?

musharaf-nocodb replied:
no i thinks its prettier

musharaf-nocodb replied:
**fixed in below Commit**

[fix : removed unwanted ! - important](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)
[b862466](https://github.com/nocodb/nocodb/pull/7114/commits/b862466785ff4be99629740e90baa00bb3b14731)

wingkwong reviewed (commented):
Whenever you apply class in `NcTooltip`, the classes are duplicated.

<img width="612" alt="image" src="https://github.com/nocodb/nocodb/assets/35857179/fdfee114-0388-436f-9c33-4bd4d8d41f7c">

mustafapc19 commented:
@wingkwong That is known issue with NcTooltip. I'll fix in another PR 👍

wingkwong reviewed (commented):
Missing tooltip text in formula. Probably other virtual fields too.

![image](https://github.com/nocodb/nocodb/assets/35857179/d66b538b-5ad5-4117-885c-525d41581f3d)

musharaf-nocodb commented:
> `[fix : fixed tooltip for virtual cell](/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21) `
> ` [e84c468](/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)`

**fixed in below commit**

[fix : fixed tooltip for virtual cell](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)
[e84c468](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)

musharaf-nocodb commented:
> Missing tooltip text in formula. Probably other virtual fields too.
> 
> ![image](https://private-user-images.githubusercontent.com/35857179/286885647-d66b538b-5ad5-4117-885c-525d41581f3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNDQyMzIsIm5iZiI6MTcwMTM0MzkzMiwicGF0aCI6Ii8zNTg1NzE3OS8yODY4ODU2NDctZDY2YjUzOGItNWFkNS00MTE3LTg4NWMtNTI1ZDQxNTgxZjNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDExMzIxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0NGI3YWY4ZjZmOGI1YTRkZGY3ZDQyZWFjMDE4YzhmNmIxY2VlMmQxMTIwYTVmMzhmZWZhNzExOTRiNDllMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f-7cIt5OP_PPcOlft5tNA0a6uGpysq7P4u8UvYyftJY)

**Fixed in below commit** 

[fix : fixed tooltip for virtual cell](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)
[e84c468](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)

musharaf-nocodb commented:
> Expanded form used to have left and right padding which seems to be removed. Intentional? New UI: <img alt="image" width="1515" src="https://private-user-images.githubusercontent.com/35809690/286823610-0ec8b227-7507-4509-9192-87e837f44d8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNDQyMzEsIm5iZiI6MTcwMTM0MzkzMSwicGF0aCI6Ii8zNTgwOTY5MC8yODY4MjM2MTAtMGVjOGIyMjctNzUwNy00NTA5LTkxOTItODdlODM3ZjQ0ZDhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDExMzIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5OWI0NzIwYzFlNTI5MjJlOTdmZDFmNTBkNjIxYzg1YmE1Yzk0ZWY0NjNkMTdmMjFiNmNhOTVmZDRmNDhmMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KcjY6V-fDdkbslX7ND-dBdakqaHq0dmGXG3WpjiI4Kw"> Old UI: <img alt="image" width="1493" src="https://private-user-images.githubusercontent.com/35809690/286823733-d29c97d6-543a-4f84-b3f3-0f400515022e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNDQyMzEsIm5iZiI6MTcwMTM0MzkzMSwicGF0aCI6Ii8zNTgwOTY5MC8yODY4MjM3MzMtZDI5Yzk3ZDYtNTQzYS00Zjg0LWIzZjMtMGY0MDA1MTUwMjJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDExMzIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliYTI4OWM2NTE3ODQ2ZjZjODAxZTY1MTZkMGI3ZTFhYWYyZGY3OTBmYjcxMjM1YTczMmM2MGVmYzkwMDE4NzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FVtdlYILnQhnfUmF-Ac_h68HjkqhXvagmJrD6OMgnVo">

**Fixed in below commit**

[fix : fixed tooltip for virtual cell](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)
[e84c468](https://github.com/nocodb/nocodb/pull/7114/commits/e84c4684d660cf2acfffa9b86e223b4c63480c21)
